### PR TITLE
Fix maximum call stack exceeded + handle ENOTFOUND

### PR DESCRIPTION
**Pull request by getvega**
- fix an error when logging an object with circular references
- cf https://github.com/BrightCrowd/winston-loggly/commit/b8714d4648759133c004eb13f5fa0289c129b1e3
- do not crash on internet disconnect

### CODE REVIEW

File: lib/winston-logstash.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ Logstash.prototype.connect = function () {
 
     self.socket = null;
 
-    if (!ECONNREFUSED_REGEXP.test(err.message)) {
+    if (err.message.indexOf('E') !== 0) {
```
Comment (djMax) [inline lib/winston-logstash.js]: This seems a bit odd... Is the source of these errors such that "starting with E" is sufficiently specific not to catch some user-created error that starts with E?

**Comment by djMax** [comment]
I don't think yarn.lock should be in here.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "winston-logstash",
-  "version": "0.3.0",
+  "name": "winston-logstash-cubyn",
```
Comment (djMax) [inline package.json]: Well we definitely can't merge it now. :)

File: lib/winston-logstash.js [hunk 3]
```diff
@@ -75,6 +74,7 @@ winston.transports.Logstash = Logstash;
 Logstash.prototype.name = 'logstash';
 
 Logstash.prototype.log = function (level, msg, meta, callback) {
+  meta = meta !== null && meta !== undefined && !(meta instanceof Error) ? cycle.decycle(meta) : meta || null;
```
Comment (djMax) [inline lib/winston-logstash.js]: cycle.decycle is in winston itself, is there some reason it has to be duplicated here?

File: lib/winston-logstash.js (hunk 1 above)
Comment (getvega) [inline lib/winston-logstash.js]: Being honest we did not find the real problem yet. We have a very tricky Internet - and we keep having `winston-logstash` crashing our whole stack when Internet is down, even after setting `max_connect_retries: -1`.

File: lib/winston-logstash.js (hunk 3 above)
Comment (getvega) [inline lib/winston-logstash.js]: Agreed, winston does it already, but we kept having max stack exceeded errors...

Sorry for this shitty PR, I will recreate it by the end of the week, including a test on circular references.

**Comment by getvega** [comment]
Sorry for this PR - I will recreate a clean one in a few days

**Pull request closed without merging**